ompi/errhandler/errhandler_predefined.c

```diff
@@ -177,21 +177,30 @@ static void backend_fatal_aggregate(char *type,
                                     char *name, int *error_code,
                                     va_list arglist)
 {
-    char *arg, *prefix, *err_msg = "Unknown error";
-    bool err_msg_need_free = false;
+    char *arg = NULL, *prefix = NULL, *err_msg = NULL;
+    const char* const unknown_error_code = "Error code: %d (no associated error message)";
+    const char* const unknown_error = "Unknown error";
 
     arg = va_arg(arglist, char*);
     va_end(arglist);
 
-    asprintf(&prefix, "[%s:%d]", ompi_process_info.nodename,
-             (int) ompi_process_info.pid);
+    if (asprintf(&prefix, "[%s:%d]",
+                 ompi_process_info.nodename,
+                 (int) ompi_process_info.pid) == -1) {
+        // non-fatal, we could still go on to give useful information here...
+        opal_output(0, "%s", "Could not write node and PID to prefix");
+        opal_output(0, "Node: %s", ompi_process_info.nodename);
+        opal_output(0, "PID: %d", (int) ompi_process_info.pid);
+    }
 
     if (NULL != error_code) {
         err_msg = ompi_mpi_errnum_get_string(*error_code);
         if (NULL == err_msg) {
-            err_msg_need_free = true;
-            asprintf(&err_msg, "Error code: %d (no associated error message)",
-                     *error_code);
+            if (asprintf(&err_msg, unknown_error_code,
+                         *error_code) == -1) {
```

Comment:
Will a failed `asprintf()` leave NULL in `err_msg`?  Or could it be some random value (that `free(err_msg)`, below, will choke on)?